perception/sub8_perception/src/sub8_perception/start_gate.cpp

```diff
@@ -10,66 +10,60 @@ Sub8StartGateDetector::~Sub8StartGateDetector() {}
+            for(auto &gate1 : gate_line_buffer) {
+                for (auto &gate2 : gate1) {
+                    cv::Point2i s = gate2[1] -gate2[0];
                     cv::Point2i center = cv::Point2i(s.x / 2 + s.x, s.y / 2 + s.y);
-                    // std::cout << center << std::endl;
                     accX(center.x);
                     accY(center.y);
                     accSizeX(s.x);
                     accSizeY(s.y);
                 }
             }
-            // std::cout << mean(acc) << std::endl;
-            // for (int i = 0; i < gate_line_buffer.size(); i++) {
-            //     for (int i = 0; i < gate_line_buffer[0].size(); i++) {
-            // cv::Point2i s = gate_line_buffer[0][i][1] - gate_line_buffer[0][i][0];
-            // cv::Point2i center = cv::Point2i(s.x / 2 + s.x, s.y / 2 + s.y);
-            // std::cout << center << std::endl;
-
-            //         //Make bigger buffer. After like 30 calculate mean and average. Outlier stuff.
-            //     }
-            // }
-            for (size_t i = 0; i < gate_line_buffer[29].size(); i++) {
-                cv::Point2i s = gate_line_buffer[29][i][1] - gate_line_buffer[29][i][0];
+
+            //Look at one of the elements (gate_line_buffer.capacity()-1) in the circular buffer and check it's statistics and delete if necessary
+            for (size_t i = 0; i < gate_line_buffer[gate_line_buffer.capacity()-1].size(); i++) {
+                cv::Point2i s = gate_line_buffer[gate_line_buffer.capacity()-1][i][1] - gate_line_buffer[gate_line_buffer.capacity()-1][i][0];
                 cv::Point2i center = cv::Point2i(s.x / 2 + s.x, s.y / 2 + s.y);
-                // std::cout << center << std::endl;
+
+                //If position of the element is greater than one deviation from the mean, then delete
                 if (abs(center.x - mean(accX)) > sqrt(variance(accX)) ||
                         abs(center.y - mean(accY) > sqrt(variance(accY)))) {
-                    gate_line_buffer[29].erase(gate_line_buffer[29].begin() + i);
+                    gate_line_buffer[gate_line_buffer.capacity()-1].erase(gate_line_buffer[gate_line_buffer.capacity()-1].begin() + i);
                     i--;
                     std::cout << "Deleted: " << center << std::endl;
                     continue;
                 }
+
+                //If the size of the element is greater than one deviation from the mean, then delete
                 if (abs(s.x - mean(accSizeX)) > sqrt(variance(accSizeX)) ||
                         abs(s.y - mean(accSizeY)) > sqrt(variance(accSizeY))) {
-                    gate_line_buffer[29].erase(gate_line_buffer[29].begin() + i);
+                    gate_line_buffer[gate_line_buffer.capacity()-1].erase(gate_line_buffer[gate_line_buffer.capacity()-1].begin() + i);
                     i--;
                     std::cout << "Deleted size: " << s << std::endl;
```

Comment:
`ROS_INFO_STREAM("Deleted size: " << s)` You shouldn't just use cout from code running on the robot.